src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,769 @@
+     * @return the set of features that should be installed from server.xml
+     */
+    public Set<String> getServerFeatures(File serverDirectory, boolean noPluginListedFeatures) {
+        // parse server.xml features only if there are no configured features in the pom
+        if (noPluginListedFeatures) {
+            debug("No features were listed for the plugin. Using server.xml.");
+            return getServerFeatures(serverDirectory);
+        } else {
+            debug("Features were listed for the plugin. Skipping server.xml.");
+            return new HashSet<String>();
+        }
+    }
+
+    /**
+     * Get the set of features defined in the server.xml
+     * @param serverDirectory The server directory containing the server.xml
+     * @return the set of features that should be installed from server.xml, or empty set if nothing should be installed
+     */
+    public static Set<String> getServerFeatures(File serverDirectory) {
+        Set<String> defaults = getConfigDropinsFeatures(serverDirectory, "defaults");
+        Set<String> defaultsAndServerXmlFeatures = getServerXmlFeatures(defaults, new File(serverDirectory, "server.xml"), null);
+        // add the overrides at the end since they should not be replaced by any other content
+        Set<String> overrides = getConfigDropinsFeatures(serverDirectory, "overrides");
+        return combineToSet(defaultsAndServerXmlFeatures, overrides);
+    }
+    
+    /**
+     * Gets features from the configDropins's defaults or overrides directory
+     * 
+     * @param serverDirectory
+     *            The server directory
+     * @param folderName
+     *            The folder under configDropins: either "defaults" or
+     *            "overrides"
+     * @return The set of features to install, or empty set if the folder has xml
+     *         files with featureManager sections but no features to install, or
+     *         null if there are no xml files or they have no featureManager
+     *         section
+     */
+    private static Set<String> getConfigDropinsFeatures(File serverDirectory, String folderName) {
+        File configDropinsFolder;
+        try {
+            configDropinsFolder = new File(new File(serverDirectory, "configDropins"), folderName).getCanonicalFile();
+        } catch (IOException e) {
+            // skip this directory if its path cannot be queried
+            return null;
+        }
+        File[] configDropinsXmls = configDropinsFolder.listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                return name.endsWith(".xml");
+            }
+        });
+        if (configDropinsXmls == null || configDropinsXmls.length == 0) {
+            return null;
+        }
+        Set<String> result = null;
+        for (File xml : configDropinsXmls) {
+            Set<String> features = getServerXmlFeatures(null, xml, null);
```

Comment:
For configDropins, I previously asked about the order of the parsing of the files. You said:

> For configDropins, we verified (using installUtility) that features do not get overridden, but are added to the set.

How does this handle `onConflict="replace"`? Is the replace just treated like a merge or something?

Reply:
"replace" replaces all previously parsed features, including any from "configDropins/defaults".  It does not replace features that come after it in the server.xml, and does not replace "configDropins/overrides".  This is tested in the unit test InstallFeatureUtilGetServerFeaturesTest.testCombinationWithReplace().

Reply:
What if there is a replace in `configDropins/overrides`? Will that replace everything previously found in `defaults` and `server.xml`?

Reply:
It should replace everything previously found.  Added logic to handle this case and a corresponding unit test case.